src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts

```diff
@@ -239,59 +240,88 @@ export class RgwBucketFormComponent extends CdForm implements OnInit {
    *   start and end with a lowercase letter or a number.
    */
   bucketNameValidator(): AsyncValidatorFn {
-    const rgwBucketService = this.rgwBucketService;
-    return (control: AbstractControl): Promise<ValidationErrors | null> => {
-      return new Promise((resolve) => {
-        // Exit immediately if user has not interacted with the control yet
-        // or the control value is empty.
-        if (control.pristine || control.value === '') {
-          resolve(null);
-          return;
+    return (control: AbstractControl): Observable<ValidationErrors | null> => {
+      // Exit immediately if user has not interacted with the control yet
+      // or the control value is empty.
+      if (control.pristine || control.value === '') {
+        return observableOf(null);
+      }
+      const constraints = [];
+      let errorName: string;
+      // - Bucket names cannot be formatted as IP address.
+      constraints.push(() => {
+        const ipv4Rgx = /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/i;
+        const ipv6Rgx = /^(?:[a-f0-9]{1,4}:){7}[a-f0-9]{1,4}$/i;
```

Comment:
Aren't there other places in our code where IPv4/6 needs be validated so we make this reusable? Or couldn't we replace part of this validation logic with some external validator library (I did a quick check but [no luck](https://www.npmjs.com/package/ngx-custom-validators))?

Reply:
I think we should add test for this
```js
const tests = [
'1.1.1.01',
'001.1.1.01',
]
const ipv4Rgx = /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/i;
for(let t in tests)
{
	const b = ipv4Rgx.test(tests[t])
	console.log(b)
}
```
output:
```
true
true
```
I don't think anyone will type that but who knows

Reply:
I think we should add test for this
```js
const tests = [
'1.1.1.01',
'001.1.1.01',
]
const ipv4Rgx = /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/i;
for(let t in tests)
{
	const b = ipv4Rgx.test(tests[t])
	console.log(b)
}
```
output:
```
true
true
```
I don't think anyone will type that but who knows

Reply:
@epuertat There's a `CdValidators.ip()` we are using, but I was not succesfull in making it work over here. It was actually used here previously but its actually incorrectly validating the fields (even if we enter IP address it doesn't give error). I checked the external libraries but none for the IP address.